client/config/config.go

```diff
@@ -95,3 +95,24 @@ func ReadFromClientConfig(ctx client.Context) (client.Context, error) {
 
 	return ctx, nil
 }
+
+// CreateClientConfigAtPath sets up a basic configuration structure in the given
+// directory.
+func CreateClientConfigAtPath(configPath, chainID string) error {
+	configFilePath := filepath.Join(configPath, "client.toml")
+
+	if err := os.MkdirAll(configPath, os.ModePerm); err != nil {
```

Comment:
This os.ModePerm `0777` is very permissible giving read-write-execute access to everyone. Could we perhaps make it `0740`. Please see https://cyber.orijtech.com/scsec/cosmos-hardening#file-permissions